### PR TITLE
fix(keychain): preserve credentials on failed updates (#148)

### PR DESCRIPTION
**Pull request by psimaker**
Update existing Keychain items in place and add only after a confirmed miss, so transient failures cannot erase the last valid credential. Gate APNs and Relay follow-on work on durable persistence and cover Security failures and duplicate races with deterministic tests.

## Component(s)
- [ ] go (bridge / Syncthing)
- [ ] ios (app / widget)
- [ ] notify (relay)
- [ ] docs / CI

## Testing
- [ ] `cd go && make patch && go test -tags noassets ./bridge`
- [ ] `cd notify && go test ./...`
- [ ] iOS build / `xcodebuild test`
- [ ] Not applicable

## Summary

- Update Keychain items in place and add new items only after a confirmed miss.
- Preserve valid credentials when encoding, read, Security, or duplicate-item operations fail.
- Gate APNs and Cloud Relay follow-on work on durable persistence.
- Report Relay storage failures in diagnostics, provisioning, purchase, and restore flows.
- Preserve secure-storage data and prevent false registration success.
- Add injectable effects and deterministic tests for Security failures, encoding failures, duplicate-item races, APNs failures, and Relay persistence failures.

### CODE REVIEW

File: ios/VaultSync/Services/SubscriptionManager.swift [hunk 1]
```diff
@@ -5,6 +5,190 @@ import os
...
+    static func decodeStoredValue(_ stored: String) -> LoadResult {
+        if stored.hasPrefix("[") {
+            guard let data = stored.data(using: .utf8),
+                  let ids = try? JSONDecoder().decode([String].self, from: data) else {
+                return .failed
+            }
+            return .loaded(ids)
+        }
+        return .loaded(stored.components(separatedBy: ",").filter { !$0.isEmpty })
+    }
```
[the remark this answers is not in the document]
Comment (psimaker) [reply ios/VaultSync/Services/SubscriptionManager.swift:128]: Good catch. Verified: `decodeStoredValue` currently validates the container syntax but not canonical Syncthing Device ID semantics. A malformed member can therefore reach stored-target selection and Relay request construction.

The same semantic acceptance existed in the original loader (`917a87e`); #160 moved that behavior into the typed loader but did not introduce it. This is a valid fail-closed hardening gap, but it is distinct from the scoped update-without-delete defect in #148. The existing pure validator is `SyncthingDeviceID.canonicalize`.

Tracked in #161 with deterministic JSON and legacy-format coverage plus fail-closed, no-rewrite, and no-provision semantics. Under the explicit owner scope decision, #160 stays limited to #148, and this review thread is resolved through that follow-up.
